File: app/src/main/java/com/example/yachae_sqlite/DBOpenHelper.java
package com.example.calendar;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

public class DBOpenHelper extends SQLiteOpenHelper {

    public static final String DB_NAME="EVENTS_DB";
    public static final int DB_VERSION=1;
    public static final String EVENT_TABLE_NAME="eventstable";
    public static final String EVENT="event";
    public static final String TIME="time";
    public static final String DATE="date";
    public static final String MONTH="month";
    public static final String YEAR="year";

    private static final  String CREATE_EVENTS_TABLE="create table "+EVENT_TABLE_NAME+
        "(ID INTEGER PRIMARY KEY AUTOINCREMENT, "+EVENT+" TEXT, "+DATE+" TEXT, "+MONTH+" TEXT, "
            +YEAR+" TEXT)";

    private static final  String DROP_EVENTS_TABLE="DROP TABLE IF EXISTS "+EVENT_TABLE_NAME;


    public DBOpenHelper(@Nullable Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_EVENTS_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(DROP_EVENTS_TABLE);
        onCreate(db);
    }

    public void SaveEvent(String event, String date,String month,String year,SQLiteDatabase database){
        ContentValues contentValues= new ContentValues();
        contentValues.put(EVENT,event);
        contentValues.put(DATE,date);
        contentValues.put(MONTH,month);
        contentValues.put(YEAR,year);
        database.insert(EVENT_TABLE_NAME,null,contentValues);

    }

    public Cursor ReadEvents(String date, SQLiteDatabase database){
     String [] Projections={EVENT,TIME,DATE,MONTH,YEAR};
     String Selection=DATE+"=?";
     String [] SelectionArgs = {date};

     return database.query(EVENT_TABLE_NAME,Projections,Selection,SelectionArgs,null,null,null);
    }

    public Cursor ReadEventsperMonth(String month,String year, SQLiteDatabase database){
        String [] Projections={EVENT,DATE,MONTH,YEAR};
        String Selection=MONTH+"=? and "+YEAR+"=?";
        String [] SelectionArgs = {month, year};

        return database.query(EVENT_TABLE_NAME,Projections,Selection,SelectionArgs,null,null,null);
    }
}

File: app/src/main/java/com/example/yachae_sqlite/DBOpenHelper.kt
package com.example.yachae_sqlite

import android.database.sqlite.SQLiteOpenHelper
import com.example.yachae_sqlite.DBOpenHelper
import android.database.sqlite.SQLiteDatabase
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import com.google.android.material.chip.Chip

class DBOpenHelper(
    customCalendarActivity: CustomCalendarActivity,
    chipName: String,
    context: Context?,
    i: Int
) : SQLiteOpenHelper(context, "challenge.db", null,1) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE chipName (milk TEXT, dairyproduct TEXT, vegetable TEXT, egg TEXT,fish text,meat text, fowls text)")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        //db.execSQL(DROP_EVENTS_TABLE)
        //onCreate(db)
    }

    fun insertData(chipName: String?): Boolean {
        val db = this.writableDatabase
        val contentValues = ContentValues()
        contentValues.put("chipName", chipName)
        val result = db.insert("chipName", null, contentValues)
        if (result.equals(-1))
            return false
        else
            return true
    }
/*
    fun SaveEvent(
        event: String?,
        date: String?,
        month: String?,
        year: String?,
        database: SQLiteDatabase
    ) {
        val contentValues = ContentValues()
        contentValues.put(EVENT, event)
        contentValues.put(DATE, date)
        contentValues.put(MONTH, month)
        contentValues.put(YEAR, year)
        database.insert(EVENT_TABLE_NAME, null, contentValues)
    }*/

/*
    fun SaveEvent(
        event: String?,
        date: String?,
        month: String?,
        year: String?,
        database: SQLiteDatabase
    ) {
        val contentValues = ContentValues()
        contentValues.put(EVENT, event)
        contentValues.put(DATE, date)
        contentValues.put(MONTH, month)
        contentValues.put(YEAR, year)
        database.insert(EVENT_TABLE_NAME, null, contentValues)
    }

    fun ReadEvents(date: String, database: SQLiteDatabase): Cursor {
        val Projections = arrayOf(EVENT, TIME, DATE, MONTH, YEAR)
        val Selection = DATE + "=?"
        val SelectionArgs = arrayOf(date)
        return database.query(
            EVENT_TABLE_NAME,
            Projections,
            Selection,
            SelectionArgs,
            null,
            null,
            null
        )
    }

    fun ReadEventsperMonth(month: String, year: String, database: SQLiteDatabase): Cursor {
        val Projections = arrayOf(EVENT, DATE, MONTH, YEAR)
        val Selection = MONTH + "=? and " + YEAR + "=?"
        val SelectionArgs = arrayOf(month, year)
        return database.query(
            EVENT_TABLE_NAME,
            Projections,
            Selection,
            SelectionArgs,
            null,
            null,
            null
        )
    }

    companion object {
        const val EVENT_TABLE_NAME = "eventstable"
        const val EVENT = "event"
        const val TIME = "time"
        const val DATE = "date"
        const val MONTH = "month"
        const val YEAR = "year"
        private const val DROP_EVENTS_TABLE = "DROP TABLE IF EXISTS " + EVENT_TABLE_NAME
    }*/
}